haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java

```diff
@@ -94,36 +94,43 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
      */
     public record TopVowelAndConsonant(char vowel, char consonant) {};
 
-    public TopVowelAndConsonant topVowelAndConsonant()
-    {
-        // TODO: Find all of the alphabetic letters from this.getHaikuAsChars(), convert them to lowercase,
-        // TODO: count the Character values in a Map and create a List of top 26 occurrences in
-        // TODO: Map.Entry<Character, Long> instances sorted in reverse order by count
-        // Hint: Look at filter, map, mapToObj, collect, Collectors.groupingBy, Collectors.counting
-        // Hint: Also look at entrySet, stream, sorted, Map.Entry.comparingByValue, Comparator.reverseorder, toList
-        List<Map.Entry<Character, Long>> entries = null;
-
-        // TODO: Find the top vowel
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top vowel Character value using getKey on Map.Entry.
-        char topVowel = 'a';
-
-        // TODO: Find the top consonant
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top consonant Character value using getKey on Map.Entry.
-        char topConsonant = 'b';
+    public TopVowelAndConsonant topVowelAndConsonant() {
+        // Find all of the alphabetic letters from the haiku, convert them to lowercase,
+        // and count the Character values in a Map.
+        Map<Character, Long> charCounts = this.getHaikuAsChars()
+                .stream()
```

Comment:
IntStream does not have the method .stream() so this code doesn't compile. You already have an IntStream, you just need to convert it into a stream of character objects. I would take a look at the IntStream.mapToObj([IntFunction](https://docs.oracle.com/javase/8/docs/api/java/util/function/IntFunction.html)<? extends U> mapper) function.

https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html

Reply:
I worked on the solution as per your suggestions. Thanks for the link, It was helpful in understanding the function better..